moped-etl/arcgis/components_to_agol.py

```diff
@@ -100,28 +110,112 @@ def main():
                 # a multi-line geometry
                 line_geometry["coordinates"] = [line_geometry["coordinates"]]
             line_feature = make_esri_feature(
-                esri_geometry_key="paths", geometry=line_geometry, attributes=component
+                esri_geometry_key="paths",
+                geometry=line_geometry,
+                attributes=component,
             )
             all_features["combined"].append(line_feature)
         else:
             all_features["lines"].append(feature)
             all_features["combined"].append(feature)
 
-    for feature_type in ["points", "lines", "combined"]:
-        logger.info(f"Processing {feature_type} features...")
-        features = all_features[feature_type]
+    return all_features
 
-        logger.info("Deleting all existing features...")
-        delete_features(feature_type)
 
-        logger.info(
-            f"Uploading {len(features)} features in chunks of {UPLOAD_CHUNK_SIZE}..."
-        )
-        for feature_chunk in chunks(features, UPLOAD_CHUNK_SIZE):
-            logger.info("Uploading chunk....")
-            add_features(feature_type, feature_chunk)
+def main(args):
+    logger.info("Getting token...")
+    get_token()
+
+    if args.full:
+        logger.info("Getting component features from all projects...")
+        data = make_hasura_request(
+            query=COMPONENTS_QUERY_BY_LAST_UPDATE_DATE,
+            variables={"where": {}},
+        )["component_arcgis_online_view"]
+
+        all_features = make_all_features(data)
```

Comment:
minor thing, but it looks like you could cut back on some dupe code by making the calls to `make_hasura_request` and `make_all_features` before you do the branching logic by assigning the `where` statement to a var

Reply:
nice - yes, even better! 🧹